mongodb/connection.go

```diff
@@ -84,18 +85,33 @@ func (ms *MongoConnection) ListCollectionsFor(ctx context.Context, database stri
 	collectionNames, err := ms.
 		client.
 		Database(database).
-		ListCollectionNames(ctx, bson.D{{"name", bson.D{{"$ne", ""}}}})
+		ListCollectionNames(ctx, bson.D{{Key: "name", Value: bson.D{{Key: "$ne", Value: ""}}}})
 	if err != nil {
 		return nil, err
 	}
 
 	return collectionNames, nil
 }
 
+func (ms *MongoConnection) d() *mongo.Database {
+	return ms.client.Database(ms.database)
+}
+
 func (ms *MongoConnection) C(collection string) *Collection {
-	return NewCollection(ms.client.Database(ms.database).Collection(collection))
+	return NewCollection(ms.d().Collection(collection))
 }
 
 func (ms *MongoConnection) DropDatabase(ctx context.Context) error {
-	return ms.client.Database(ms.database).Drop(ctx)
+	return ms.d().Drop(ctx)
+}
+
+// RunCommand executes the given command against the configured database.
+// This is provided for tests only and no values are returned
+func (ms *MongoConnection) RunCommand(ctx context.Context, runCommand interface{}) error {
+	res := ms.d().RunCommand(ctx, runCommand)
+
+	if res.Err() != nil || res.Err() == mongo.ErrNoDocuments {
+		return res.Err()
+	}
```

Comment:
Can this be simplified to:
return res.Err()

Reply:
I thought we would like to handle `ErrNoDocuments` as this can be returned by the operation if it was successful but did not return any document. Since we don't really care about the returned object (this is only provided for test) maybe we want to abstract this and only return an `error` if something went wrong?

Reply:
Agreed to simplify this